helusers/models.py

```diff
@@ -131,3 +132,36 @@ def __str__(self):
     class Meta:
         abstract = True
         ordering = ('id',)
+
+
+class OIDCBackChannelLogoutEventManager(models.Manager):
+    def logout_token_received(self, logout_token):
+        sub = logout_token.claims.get("sub", "")
+        sid = logout_token.claims.get("sid", "")
+
+        try:
+            with transaction.atomic():
+                self.create(iss=logout_token.issuer, sub=sub, sid=sid)
+        except IntegrityError:
+            pass
+
+    def is_session_terminated_for_token(self, token):
+        sid = token.claims.get("sid")
+        if sid:
+            return self.filter(iss=token.issuer, sid=sid).exists()
+
+        return False
+
+
+class OIDCBackChannelLogoutEvent(models.Model):
```

Comment:
An idea  / question about naming.

Is it conceivable that in the future, we might have other "channels" through which API tokens could be invalidated? This model name ties the event to the specific OIDC endpoint, but could it be named more generically, like "TokenInvalidationEvent" or something? Not sure, just a question. Of course it might be better to think about this later if it becomes relevant.

Reply:
I would leave such pondering later, if/when it becomes relevant.